services/provider/server/server.go

```diff
@@ -690,3 +705,34 @@ func (s *OCSProviderServer) ReportStatus(ctx context.Context, req *pb.ReportStat
 
 	return &pb.ReportStatusResponse{}, nil
 }
+
+// PeerBlockPool RPC call to send the bootstrap secret for the pool
+func (s *OCSProviderServer) PeerBlockPool(ctx context.Context, req *pb.PeerBlockPoolRequest) (*pb.PeerBlockPoolResponse, error) {
+
+	if err := s.cephRBDMirrorManager.Create(ctx); err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to create RBDMirror instance: %v", err)
+	}
+
+	// enable mirroring on blockPool in the req
+	if err := s.cephBlockPoolManager.EnableBlockPoolMirroring(ctx, req.BlockPoolName); err != nil {
+		if kerrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "Failed to find CephBlockPool resource %s: %v", req.BlockPoolName, err)
+		}
+		return nil, status.Errorf(codes.Internal, "Failed to enable mirroring for CephBlockPool resource %s: %v", req.BlockPoolName, err)
+	}
+
+	// create and set secret ref on the blockPool
+	if err := s.cephBlockPoolManager.SetBootstrapSecretRef(
+		ctx,
+		req.BlockPoolName,
+		map[string][]byte{
+			"pool":  req.Pool,
+			"token": req.Token,
```

Comment:
will you get a request w/ single pool name but different bytes on `pool` and `token`?

Reply:
Yes, when we have more than 2 clusters, the token will be different for different ceph clusters